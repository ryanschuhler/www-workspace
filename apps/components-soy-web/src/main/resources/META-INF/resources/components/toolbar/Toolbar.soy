{namespace Toolbar}

/**
 * Toolbar
 */
{template .render}
  {@param? elementClasses: string}
  {@param? items: list<?>}

  <header class="mdc-toolbar {$elementClasses}">
    {if $items}
      <div class="mdc-toolbar__row">
        {foreach $item in $items}
          {call ._toolbarSection}
            {param align: $item.align /}
            {param content: $item.content /}
            {param elementClasses: $item.elementClasses /}
          {/call}
        {/foreach}
      </div>
    {/if}
  </header>
{/template}

/**
 * MaterialModal
 */
{template ._toolbarSection}
  {@param? align: string}
  {@param? content: html|string}
  {@param? elementClasses: string}

  {let $attributes kind="attributes"}
    class="mdc-toolbar__section
    {if $align == 'start'}
      {sp}mdc-toolbar__section--align-start
    {elseif $align == 'end'}
      {sp}mdc-toolbar__section--align-end
    {/if}

    {if isNonnull($elementClasses)}
      {sp}{$elementClasses}
    {/if}
    "
  {/let}

  <section {$attributes}>
    {$content}
  </section>
{/template}