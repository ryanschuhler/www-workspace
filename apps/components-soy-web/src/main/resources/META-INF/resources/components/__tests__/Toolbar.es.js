'use strict';

import Toolbar from '../toolbar/Toolbar.es';

const items = [
  {
    align: 'start',
    content: '<h1>Item One</h1>',
    elementClasses: 'item-1'
  },
  {
    align: 'end',
    content: '<h1>Item Two</h1>',
    elementClasses: 'item-2'
  },
  {
    align: '',
    content: '<h1>Item Three</h1>',
    elementClasses: 'item-3'
  }
];

let component;

describe('Toolbar', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  beforeAll(() => {
		window.Liferay = {};
  });

  it('should render with items', () => {
    component = new Toolbar({
      items: items
    });

    expect(component).toMatchSnapshot();
  });
});