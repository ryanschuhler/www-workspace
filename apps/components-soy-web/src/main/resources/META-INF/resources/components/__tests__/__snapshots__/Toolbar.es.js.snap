// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should render with items 1`] = `
<header class="mdc-toolbar undefined">
  <div class="mdc-toolbar__row">
    <section class="mdc-toolbar__section mdc-toolbar__section--align-start item-1">&lt;h1&gt;Item One&lt;/h1&gt;</section>
    <section class="mdc-toolbar__section mdc-toolbar__section--align-end item-2">&lt;h1&gt;Item Two&lt;/h1&gt;</section>
    <section class="mdc-toolbar__section item-3">&lt;h1&gt;Item Three&lt;/h1&gt;</section>
  </div>
</header>
`;
